src/ui/pages/SwapForm/Quotes/QuoteList.tsx

```diff
@@ -0,0 +1,286 @@
+import { emitter } from 'src/ui/shared/events';
+import { useLocation } from 'react-router-dom';
+import { useTransactionFee } from '../../SendTransaction/TransactionConfiguration/useTransactionFee';
+import { FeeDescription } from './FeeDescription';
+import type { FeeTier } from './FeeTier';
+import * as styles from './styles.module.css';
+
+const QUOTE_GRIP_TEMPLATE_COLUMNS = '1fr 1fr 40px';
+
+function QuoteNetworkFee({
+  transaction,
+  chain,
+  configuration,
+}: {
+  transaction: NonNullable<Quote['transaction']>;
+  chain: Chain;
+  configuration: CustomConfiguration;
+}) {
+  const { currency } = useCurrency();
+  const { data: chainGasPrices = null } = useGasPrices(chain);
+  const transactionFee = useTransactionFee({
+    address: transaction.from,
+    transaction: {
+      ...omit(transaction, ['chain_id']),
+      chainId: transaction.chain_id,
+    },
+    chain,
+    onFeeValueCommonReady: null,
+    networkFeeConfiguration: configuration.networkFee,
+    chainGasPrices,
+  });
+
+  return (
+    <span>
+      {transactionFee.costs?.feeValueFiat
+        ? formatCurrencyValue(
+            transactionFee.costs?.feeValueFiat,
+            'en',
+            currency
+          )
+        : 'N/A'}
+    </span>
+  );
+}
+
+function Quote({
+  quote,
+  receiveAsset,
+  configuration,
+}: {
+  quote: Quote;
+  receiveAsset: Asset;
+  configuration: CustomConfiguration;
+}) {
+  const { currency } = useCurrency();
+  const chain = createChain(quote.output_chain);
+
+  const receiveAmount = getCommonQuantity({
+    asset: receiveAsset,
+    chain,
+    baseQuantity: quote.output_amount_estimation,
+  }).times(receiveAsset.price?.value || 0);
+
+  return (
+    <HStack
+      gap={0}
+      alignItems="center"
+      style={{ gridTemplateColumns: QUOTE_GRIP_TEMPLATE_COLUMNS }}
+    >
+      <VStack gap={0} style={{ justifyItems: 'start' }}>
+        <UIText kind="small/accent">
+          {formatCurrencyValue(
+            receiveAsset.price?.value ? receiveAmount : 'N/A',
+            'en',
+            currency
+          )}
+        </UIText>
+        {quote.enough_allowance ? (
+          <HStack gap={4} alignItems="center">
+            <TickIcon />
+            <UIText kind="caption/regular" style={{ whiteSpace: 'nowrap' }}>
+              Approved for {quote.contract_metadata?.name}
+            </UIText>
+          </HStack>
+        ) : null}
+      </VStack>
+      <VStack gap={0} style={{ justifyItems: 'start' }}>
+        <UIText kind="small/accent">
+          {quote.transaction ? (
+            <React.Suspense fallback={<CircleSpinner />}>
+              <QuoteNetworkFee
+                chain={chain}
+                transaction={quote.transaction}
+                configuration={configuration}
+              />
+            </React.Suspense>
+          ) : (
+            noValueDash
+          )}
+        </UIText>
+        {quote.enough_allowance ? <div style={{ height: 16 }} /> : null}
```

Comment:
heh:)) simply remove `alignItems="center"` from the parent HStack above :)

Reply:
Or set `align-self: start` to this UIText

Reply:
Hm, but in case of unapproved quote this text will be alighed to the top and the left value will be centered. 
And I want centered value if not approved and aligned to the top if approved. This is possible to do with another layer of HStack, but I didn't want to create it to maintain shared style for columns.